### PR TITLE
TECH Allow to consume multiple routing keys with multiple handlers

### PR DESCRIPTION
**Pull request by lledey**
### Jira issue

https://transcovo.atlassian.net/browse/TRAN-

### Purpose of this PR

...

### Configuration change

...

### Checks

- [ ] Documentation of environment variables is very clear, even for a newcomer
- [ ] All relevant usecases of are tested
- [ ] These changes won't cause memory / CPU problems even if business grows by a factor of 100
- [ ] It's ok if these changes go to production and then are rolled-back

### Linked PRs:

- ...

### CODE REVIEW

File: lib/worker.js [hunk 1]
```diff
@@ -11,29 +11,36 @@ const DEFAULT_EXCHANGE_TYPE = 'topic';
...
+ * @param {Number} [options.heartbeat] to override default heartbeat
+ * @param {Number} [options.taskTimeout] to override default task timeout
+ * @param {Number} [options.processExitTimeout] to override default process exit timeout
+ * @param {Number} [options.channelPrefetch] to override default channel prefetch value
  * @returns {Object} a worker instance with connection, channel, and listen/close functions
  */
-function createWorker(handler, config, options = {}) {
-  const workerConfig = applyConfiguration(Object.assign({}, { handler }, config));
+function createWorker(handlers, config, options = {}) {
+  const workerConfig = applyConfiguration(Object.assign({}, { handlers }, config));
```
Comment (hubertmine) [inline lib/worker.js]: validateConfig and validateOptions ? I feel that apply is confusing

File: README.md [hunk 2]
```diff
@@ -14,22 +14,20 @@ Example:
...
+    }], {
+      workerName: 'my worker',
+      amqpUrl: 'amqp://guest:guest@localhost:5672',
+      exchangeName: 'bus',
+      queueName: 'test.test_watcher'
+    }, {
+      channelPrefetch: 50,
+      taskTimeout: 30000,
+      processExitTimeout: 3000
+    });
```
Comment (rossille) [inline README.md]: Problem: this is an example of how to use `createWorkers` with the `createWorker` name

File: lib/schema.js [hunk 3]
```diff
@@ -8,23 +8,39 @@ const DEFAULT_TASK_TIMEOUT = 30000;
...
 
 const optionsSchema = Joi.object({
-  validator: Joi.func().default(_.noop),
   heartbeat: Joi.number().positive().default(DEFAULT_HEARTBEAT),
   taskTimeout: Joi.number().positive().default(DEFAULT_TASK_TIMEOUT),
   processExitTimeout: Joi.number().positive().default(DEFAULT_PROCESS_TIMEOUT),
   channelPrefetch: Joi.number().positive().default(DEFAULT_PREFETCH)
 });
 
+const legacyConfigurationSchema = Joi.object({
```
Comment (rossille) [inline lib/schema.js]: the "legacy" prefix is less explicit than the "createWorker" prefix. Maybe put all the "createWorker" stuff in a file and all the "createWorkers" stuff in another file, so we can simpliy say "configurationSchema" everywhere ?

File: README.md [hunk 4]
```diff
@@ -2,33 +2,54 @@
...
+```
+
+Or for a single handler/routing key:
```
Comment (rossille) [inline README.md]: you need to say that it's deprecated

File: lib/createWorker.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+} = require('./schema/createWorker.schema');
+
+/**
+ * Formats the createWorker parameters to call createWorkers
+ * @param   {function} handler The handler function
+ * @param   {Object} config The configuration
+ * @param   {Object} options The options
+ * @returns {Object} The formatted object
+ */
+function formatLegacyParameters(handler, config, options) {
```
Comment (rossille) [inline lib/createWorker.js]: There is still the "legacy" word in the name

File: README.md (hunk 4 above)
Comment (lledey) [inline README.md]: Removed from the README

**Pull request merged**